census_view.cpp

```diff
@@ -1669,6 +2709,81 @@ void CensusView::UponDeleteCells(wxCommandEvent&)
     status() << "Delete: " << timer.stop().elapsed_msec_str() << std::flush;
 }
 
+void CensusGridView::UponDeleteCells(wxCommandEvent&)
+{
+    auto const n_items = grid_table_->GetRowsCount();
+    auto selection = grid_window_->GetSelectedRows();
+    auto const n_sel_items = lmi::ssize(selection);
+    LMI_ASSERT(n_sel_items == lmi::ssize(selection));
+    // This handler should have been disabled if no cell is selected.
+    LMI_ASSERT(0 < n_sel_items);
+
+    if(n_items == n_sel_items)
+        {
+        warning()
+            << "Cannot delete all cells. A census must always"
+            << " contain at least one cell."
+            << LMI_FLUSH
+            ;
+        return;
+        }
+
+    LMI_ASSERT(n_sel_items < n_items);
+
+    std::ostringstream oss;
+    oss
+        << "Irrevocably delete "
+        << n_sel_items
+        << " of "
+        << n_items
+        << " cells?"
+        ;
+    int yes_or_no = wxMessageBox
+        (oss.str()
+        ,"Confirm deletion"
+        ,wxYES_NO | wxICON_QUESTION
+        );
+    if(wxYES != yes_or_no)
+        {
+        return;
+        }
+
+    wxBusyCursor reverie;
+    Timer timer;
+
+    const auto cursor_col = grid_window_->GetGridCursorCol();
+
+    std::sort(selection.begin(), selection.end());
+
+    LMI_ASSERT(lmi::ssize(cell_parms()) == n_items);
+
+    for(int j = n_sel_items - 1; 0 <= j; --j)
+        {
+        cell_parms().erase(selection[j] + cell_parms().begin());
+        }
+    LMI_ASSERT(lmi::ssize(cell_parms()) == n_items - n_sel_items);
+
+        {// Control the grid update locker life time.
```

Comment:
```suggestion
        { // Begin wxGridUpdateLocker scope.
```
(for consistency with other existing similar comments).